pg_lakehouse/src/schema/cell.rs

```diff
@@ -527,28 +527,34 @@ where
                     PgOid::from(oid),
                 )),
             },
-            pg_sys::TEXTOID | pg_sys::VARCHAROID | pg_sys::BPCHAROID => match self.data_type() {
-                DataType::Utf8 => match self.get_primitive_value::<StringArray>(index)? {
-                    Some(value) => Ok(Some(Cell::String(value.to_string()))),
-                    None => Ok(None),
-                },
-                DataType::LargeUtf8 => match self.get_primitive_value::<LargeStringArray>(index)? {
-                    Some(value) => Ok(Some(Cell::String(value.to_string()))),
-                    None => Ok(None),
-                },
-                DataType::Binary => match self.get_binary_value::<BinaryArray>(index)? {
-                    Some(value) => Ok(Some(Cell::String(value))),
-                    None => Ok(None),
-                },
-                DataType::LargeBinary => match self.get_binary_value::<LargeBinaryArray>(index)? {
-                    Some(value) => Ok(Some(Cell::String(value))),
-                    None => Ok(None),
-                },
-                unsupported => Err(DataTypeError::DataTypeMismatch(
-                    unsupported.clone(),
-                    PgOid::from(oid),
-                )),
-            },
+            pg_sys::TEXTOID | pg_sys::VARCHAROID | pg_sys::BPCHAROID | pg_sys::BYTEAOID => {
```

Comment:
Are you sure this is the right conversion for `BYTEA`? `get_binary_value` is returning a `String`, not bytes.

Reply:
No, I'm not sure! I don't fully grok what's going on in this file yet, do you have a suggestion on how to handle this?

Reply:
It looks like `bytea` is just an alias for `varlena`. `pgrx` provides a `PgVarlena` type that we can probably convert the binary into.